pkg/kv/kvpb/errors.go

```diff
@@ -1361,7 +1361,9 @@ func (e *BatchTimestampBeforeGCError) Error() string {
 
 func (e *BatchTimestampBeforeGCError) SafeFormatError(p errors.Printer) (next error) {
 	p.Printf(
-		"batch timestamp %v must be after replica GC threshold %v (r%d: %s)",
+		"Changefeed creation failed: cursor (%v) is older than the GC threshold (%v). "+
```

Comment:
This isn't the right place to make this change. As you have noticed in all the tests you've changed, this error can occur in far more scenarios than just being changefeed related.

Instead, you should find the changefeed code that encountered this error, detect the error type by using `errors.HasType`, and add this more clear error message in that area of the code by using `errors.Wrap`.

Reply:
Thanks for catching this @rafiss! I will make the requested changes.